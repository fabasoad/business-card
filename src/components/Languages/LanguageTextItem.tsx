import * as React from 'react'
import LanguageItem, { LanguageItemProps } from './LanguageItem'

export type LanguageTextItemProps = LanguageItemProps & {
  text: string
}

export default function LanguageTextItem(props: LanguageTextItemProps) {
  return (
    <LanguageItem code={props.code}>
      <span className="h5 mt-2">{props.text}</span>
    </LanguageItem>
  )
}
