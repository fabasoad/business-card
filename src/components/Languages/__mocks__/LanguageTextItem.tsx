import * as React from 'react'
import type { LanguageTextItemProps } from '../LanguageTextItem'

export default function LanguageTextItem({ code, text }: LanguageTextItemProps) {
  return (
    <div data-testid={code}>{text}</div>
  )
}
