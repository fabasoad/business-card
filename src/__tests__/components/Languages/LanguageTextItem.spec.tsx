/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import LanguageTextItem from '../../../components/Languages/LanguageTextItem'

test('should render LanguageTextItem correctly', () => {
  const wrapper: ShallowWrapper = shallow(<LanguageTextItem
    code="gb"
    text="test-text"
  />)
  expect(wrapper).toMatchSnapshot()
})
