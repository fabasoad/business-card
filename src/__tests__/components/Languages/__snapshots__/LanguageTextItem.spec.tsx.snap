// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LanguageTextItem correctly 1`] = `
<LanguageItem
  code="gb"
>
  <span
    className="h5 mt-2"
  >
    test-text
  </span>
</LanguageItem>
`;
